src/it/scala/ApiServiceISpec.scala

```diff
@@ -313,6 +327,2353 @@ class ApiServiceISpec
   private def sendMessage(session: Session, producer: MessageProducer, textMessageConfig: TextMessageConfig): Unit =
     producer.send(asTextMessage(session, textMessageConfig))
 
+  private def agentSummariesExpectedResult =
```

Comment:
I would move this JSON into seperate file in the src/it/resources directory and call it expected-agent-summaries.json. You can then replace this function with the following:
```
  private def agentSummariesExpectedResult: String = {
    val summaries = Source.fromResource("expected-agent-summaries.json").getLines().mkString
    parse(summaries) match {
      case Right(json) => json.printWith(Printer.spaces2)
      case Left(_) => ""
    }
  }
```
This will keep the test itself cleaner. You will need to update the JSON to remove the null values (see my other comment about this) and then the integration test should pass. However we should add other integration tests for the different request payload permutations.

Reply:
Done